src/CardArea.tsx

```diff
@@ -7,19 +7,56 @@ interface CardAreaProps {
   onDrop: (item: any) => void;
   children: any;
 }
-
+export enum Direction {
+  Left,
+  Right,
+  None,
+}
 const CardArea = (props: CardAreaProps) => {
-  const [{ isOver }, drop] = useDrop({
+  const [{ isOver, directionOfDrag }, drop] = useDrop({
     accept: DragTypes.CARD,
     drop: (item) => props.onDrop(item),
-    collect: (monitor) => ({
-      isOver: !!monitor.isOver(),
-    }),
-  });
 
+    collect: (monitor) => {
+      const directionOfDrag = monitor.getDifferenceFromInitialOffset()
+        ? monitor.getDifferenceFromInitialOffset()!.x > 0
+          ? Direction.Right
+          : Direction.Left
+        : Direction.None;
+      return {
+        isOver: !!monitor.isOver(),
+        directionOfDrag: directionOfDrag,
+      };
+    },
+  });
+  console.log(directionOfDrag);
   return (
-    <div className="card-area" ref={drop}>
-      <WithProps isOver={isOver}>{props.children}</WithProps>
+    <div className="card-area">
+      <div>
+        <div
+          className="left"
+          style={{
+            backgroundImage:
+              isOver && directionOfDrag === Direction.Left
+                ? "radial-gradient(lightblue 30%, transparent 70%"
+                : "",
+          }}
+        ></div>
+        <div
+          className="right"
+          style={{
+            backgroundImage:
+              isOver && directionOfDrag === Direction.Right
+                ? "radial-gradient(lightblue 30%, transparent 70%"
+                : "",
+          }}
+        ></div>
```

Comment:
Let's take the styles away from the code into .css files

JS file could look like this
```
<div
    className={classnames("highlight-area right", {
      hovered: isOver && directionOfDrag === Direction.Right
    })}
/>
```

then.. in CSS file:


.highlight-area {
   ...any styles that are shared between right and left indicators...
}
.highlight-area hovered {
   background-image: radial-gradient(lightblue 30%, transparent 70%);
}
.highlight-area right {
   ...any styles that are specific to right indicator...
}
.highlight-area left {
   ...any styles that are specific to left indicator...
}